lib/ansible/inventory/manager.py

```diff
@@ -436,30 +431,41 @@ def get_hosts(self, pattern="all", ignore_limits=False, ignore_restrictions=Fals
 
         return hosts
 
-    def _evaluate_patterns(self, patterns):
+    def _evaluate_patterns(self, patterns: list[str], hosts: list[Host]) -> list[Host]:
         """
-        Takes a list of patterns and returns a list of matching host names,
-        taking into account any negative and intersection patterns.
+        Filters a list of hosts by applying patterns.
         """
 
-        patterns = order_patterns(patterns)
-        hosts = []
+        pattern_regular, pattern_intersection, pattern_exclude = order_patterns(patterns)
+        hosts_set = set(hosts)
```

Comment:
set does add automatic unique tracking, but order is also important, which is why we used lists here (time for OrderedSet?)

Reply:
The `set` variants are just for performing `in` statement, and aren't actually used for the results. We still utilize the `hosts` variable to maintain order.

And yes, I need to get https://github.com/ansible/ansible/pull/84134 merged, but since this may be backported, I figure it could be a future change. I tried to make the code here easy to convert, by leaving explicit set operations where all modifications are done.